File: training-data/csharp/GameOfLife.cs
using System;
using System.Collections.Generic;
using System.Linq;

class GameOfLife
{
    static HashSet<(int X, int Y)> Step(HashSet<(int X, int Y)> live)
    {
        var counts = live
            .SelectMany(c => from dx in new[] { -1, 0, 1 } from dy in new[] { -1, 0, 1 } where dx != 0 || dy != 0 select (c.X + dx, c.Y + dy))
            .GroupBy(c => c)
            .Select(g => (Cell: g.Key, N: g.Count()));
        return counts.Where(x => x.N == 3 || (x.N == 2 && live.Contains(x.Cell))).Select(x => x.Cell).ToHashSet();
    }

    static void Main()
    {
        var world = new HashSet<(int, int)> { (1, 0), (2, 1), (0, 2), (1, 2), (2, 2) };
        for (int gen = 0; gen < 4; gen++)
        {
            Console.WriteLine($"gen {gen}");
            for (int y = 0; y < 6; y++)
                Console.WriteLine(string.Concat(Enumerable.Range(0, 6).Select(x => world.Contains((x, y)) ? '#' : '.')));
            world = Step(world);
        }
    }
}

File: training-data/cpp/game_of_life.cpp
#include <iostream>
#include <set>
#include <utility>

using Cell = std::pair<int, int>;
using World = std::set<Cell>;

World step(const World& live) {
    std::set<Cell> candidates;
    for (auto [x, y] : live)
        for (int dx = -1; dx <= 1; ++dx)
            for (int dy = -1; dy <= 1; ++dy) candidates.insert({x + dx, y + dy});
    World next;
    for (auto [x, y] : candidates) {
        int n = 0;
        for (int dx = -1; dx <= 1; ++dx)
            for (int dy = -1; dy <= 1; ++dy)
                if ((dx || dy) && live.count({x + dx, y + dy})) ++n;
        if (n == 3 || (n == 2 && live.count({x, y}))) next.insert({x, y});
    }
    return next;
}

int main() {
    World w{{1, 0}, {2, 1}, {0, 2}, {1, 2}, {2, 2}};  // glider
    for (int gen = 0; gen < 4; ++gen) {
        std::cout << "gen " << gen << ":\n";
        for (int y = 0; y < 6; ++y) {
            for (int x = 0; x < 6; ++x) std::cout << (w.count({x, y}) ? '#' : '.');
            std::cout << '\n';
        }
        w = step(w);
    }
}
